### PR TITLE
improve DX

### PR DESCRIPTION
**Pull request by cleemansen**
- this is ON HOLD until end of current _coding challenge_
- the final, reviewed PR must be applied to the other _R Templates_, too
- `move2` handling is currently not part of this PR

This PR should improve the development experience creating new MoveApps R Apps:

1. local app-run is controllable by `environment variables` (`.env` / `dotenv`). Therefore no need for _dirty_ R files anymore (dirty in the context of _not comittable git changes_)
2. a freshly cloned template instance _just works_
3. improved `developer_README.md`
4. added _unit test_ setup
5. improved _Docker_ instructions

### CODE REVIEW

File: developer_README.md [hunk 1]
```diff
@@ -1,29 +1,193 @@
-# Developing with the MoveApps SDK
+# MoveApps R Software Development Kit (SDK)
```
Comment (cleemansen) [inline developer_README.md:1]: add chapter about updating template fork with template updates

Comment (cleemansen) [reply developer_README.md:1]: on hold until discussed at https://trello.com/c/P9AHdzoi/835-gh-template-versioning

File: developer_README.md [hunk 2]
```diff
@@ -1,29 +1,195 @@
...
+  "createsArtifacts": true,
+```
+
+`./RFunction.R`
+
+```
+pdf(appArtifactPath("MorningReport_overviewTable.pdf"), paper = "a4r")
+```
+
+**Notice:** Only files are permitted to act as MoveApps App artifact! If your app produces a directory as an App artificat you have to bundle it eg. by zipping it. In other words: at the moment your App completes its work there must be only files present in `APP_ARTIFACTS_DIR`.
```
Comment (cleemansen) [inline developer_README.md:173]: @andreakoelzsch : sobald ihr in der anderen App einen sauberen Weg gefunden habt mehrere Dateien als ZIP-Artifact zu erzeugen könnt ihr bitte dieses Snippet hier in die Doku einfügen?

**Review by andreakoelzsch — APPROVED** [review]
I will still add the zip example tomorrow. I have also some very minor edits committed. But it is fine to be merged.

**Review by andreakoelzsch — COMMENTED** [review]
How do I have to understand the "levels"? logger.info("test abc",level="warn")?

**Comment by cleemansen** [comment]
> How do I have to understand the "levels"? logger.info("test abc",level="warn")?

All right. The adjusted comment is misleading. Will revert the change..

**Comment by cleemansen** [comment]
- [ ] Docker just works

File: RFunction.R [hunk 3]
```diff
@@ -1,10 +1,37 @@
...
+    plot(result)
+    dev.off()
+  } else {
+    logger.warn("nothing to plot")
+  }
+  # Showcase to access a file ('auxiliary files') that is 
+  # a) provided by the app-developer and 
+  # b) can be overridden by the workflow user.
+  fileName <- paste0(getAppFilePath("yourLocalFileSettingId"), "sample.txt")
+  logger.info(readChar(fileName, file.info(fileName)$size))
```
Comment (cleemansen) [inline RFunction.R:33]: Showcase LOCAL FILE hinzugefügt.

```
[INFO] Welcome to the MoveApps R SDK
[INFO] plotting to artifact: ./data/output/plot.png
[INFO] Using fallback files provided by app developer for 'yourLocalFileSettingId'.
[INFO] This file content was provided by the app developer.
```

File: developer_README.md (hunk 2 above)
Comment (andreakoelzsch) [reply developer_README.md:176]: done

**Review by andreakoelzsch — APPROVED** [review]

**Pull request merged**